### ISSUE TITLE
Kafka source connector

### ISSUE THREAD

**Issue by refset** [xtdb/crux#3690]
A Kafka source connector could be configured for generic CDC purposes, e.g. publishing database-wide events to a topic containing [`table`, `_id`, ` _row`, `_valid_time`, `_system_time`], and with special support for corrections and future-schedule updates that become valid (see related 1.x [issue](https://github.com/xtdb/xtdb/issues/676))

A source connector was previously implemented for 1.x in https://github.com/xtdb/xtdb/issues/324

**Comment by jarohen** [comment]
Closing as not planned — superseded by #2454.

A Kafka source connector is one implementation of an outbound sink. The generic capability — a supported way for an external process to receive XT's committed transactions — is what's being scoped at #2454; a built-in Kafka implementation is deliberately not part of it.

Worth raising afresh once that lands and there's an interface to build a connector against.
